urld1e4af3f63_GMIyer_myastro_tgz-pJ8-loader_MatchMakerMainJ8/src/myastro/service/matchmaker/MatchMakerCommonService.java

```diff
@@ -0,0 +1,59 @@
+package myastro.service.matchmaker;
+
+import myastro.constants.Nakshatra;
+import myastro.constants.Rashi;
+
+public class MatchMakerCommonService {
+
+	public static Integer getGirlToBoyNakshatraDistance(Nakshatra boyStar, Nakshatra girlStar) {
+		int boyNumber = boyStar.ordinal() + 1;
+		int girlNumber = girlStar.ordinal() + 1;
+		int distance = 0;
+
+		if (boyNumber > girlNumber) {
+			distance = boyNumber - (girlNumber - 1);
+		}
+		if (boyNumber < girlNumber) {
+			distance = (boyNumber + 28) - girlNumber;
+		}
+		if (boyNumber == girlNumber) {
+			distance = 1;
+		}
+		return distance;
+	}
+
+	public static Integer getGirlToBoyRashiDistance(Rashi boyRashi, Rashi girlRashi) {
+		int boyNumber = boyRashi.ordinal() + 1;
+		int girlNumber = girlRashi.ordinal() + 1;
+		int distance = 0;
+
+		if (boyNumber > girlNumber) {
+			distance = boyNumber - (girlNumber - 1);
+		}
+		if (boyNumber < girlNumber) {
+			distance = (boyNumber + 13) - girlNumber;
+		}
+		if (boyNumber == girlNumber) {
+			distance = 1;
+		}
+		return distance;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 26:39, 43:56.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.